solutions/recursion/balanced-parentheses.js

```diff
@@ -0,0 +1,36 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, 'utf-8');
+let idx = 0;
+data = data.split('\n');
+
+function readLine() {
+    idx++;
+    return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+
+function getAllBalancedParan(n) {
+    // Implement this function
+    if (n == 1) return ["()"];
+    if (n > 1) {
+        let set = new Set();
```

Comment:
@DevratAgarwal Sets were not taught to students as part of curriculum. Please go for another alternative solution